database/buildSchema/38_application_list_view.sql

```diff
@@ -72,8 +64,9 @@ CREATE FUNCTION application_list (userid int DEFAULT 0)
     LEFT JOIN application_stage_status_latest AS stage_status ON app.id = stage_status.application_id
     LEFT JOIN organisation org ON app.org_id = org.id
     LEFT JOIN assignment_list (stage_status.stage_id) ON app.id = assignment_list.application_id
-    LEFT JOIN review_list ($1) ON app.id = review_list.application_id
-    LEFT JOIN assigner_list ($1) ON app.id = assigner_list.application_id
+    LEFT JOIN review_list (stage_status.stage_id, $1) ON app.id = review_list.application_id
+    LEFT JOIN review_is_fully_assigned_list (stage_status.stage_id, 1) AS assignment_status ON app.id = assignment_status.application_id
+    LEFT JOIN assigner_list (stage_status.stage_id, assignment_status.is_fully_assigned, $1) ON app.id = assigner_list.application_id
```

Comment:
how would this work ? I think I missed this in the previous PR of filter by reviewer and assigner. i.e. what happens when you pass a table field (which is different for every row), to a method that returns a table that relies on it ?

Reply:
Not sure if I understood. I'll check with you in person in a sec

Reply:
Hope this is better to read now. But basically the idea is that `application_list_view` would generate a table with all applications (visible to the current user) and fields as `reviewer_action`, `assigner_action` if this user has any actions associated and in another fields `reviewer_usernames` and `assigner_usernames` we return associated usernames which are reviewers and assigners to this application. And now that I have said that, It seems better to use some rowLevel permissions on all these fields - to avoid retrieving sensitive information for Applicant users.

Reply:
I like the solution I just wasn't sure how postrgres handles a method call with a column key of selected records (since there is a 'where' statement within the condition.